### PR TITLE
Added servlet extension

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package org.zalando.opentracing.servlet.extension;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator;
+import io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator.HeaderEntry;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java]: Codacy Issue found: [Avoid unused imports such as 'io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator.HeaderEntry'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/StandardSpanDecoratorTest.java [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        this.client.setErrorHandler(new DefaultResponseErrorHandler() {
+            @Override
+            public void handleError(final ClientHttpResponse response) {
+                // don't throw
+            }
+        });
+    }
+
+    @Autowired
+    private MockTracer tracer;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/StandardSpanDecoratorTest.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/ServiceLoaderSpanDecorator.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.zalando.opentracing.servlet.extension;
+
+import io.opentracing.contrib.web.servlet.filter.ServletFilterSpanDecorator;
+import org.apiguardian.api.API;
+
+import java.util.ServiceLoader;
+
+import static java.util.ServiceLoader.load;
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+import static org.zalando.opentracing.servlet.extension.CompositeSpanDecorator.composite;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/ServiceLoaderSpanDecorator.java]: Codacy Issue found: [Avoid unused imports such as 'org.zalando.opentracing.servlet.extension.CompositeSpanDecorator.composite'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
+package org.zalando.opentracing.servlet.extension;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java]: Codacy Issue found: [Avoid unused imports such as 'io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/ServiceLoaderSpanDecorator.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * @see ServiceLoader
+ */
+@API(status = EXPERIMENTAL)
+public final class ServiceLoaderSpanDecorator extends ForwardingSpanDecorator {
+
+    public ServiceLoaderSpanDecorator() {
+        super(CompositeSpanDecorator.composite(load(ServletFilterSpanDecorator.class)));
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/ServiceLoaderSpanDecorator.java]: Codacy Issue found: [Unnecessary use of fully qualified name 'CompositeSpanDecorator.composite' due to existing static import 'org.zalando.opentracing.servlet.extension.CompositeSpanDecorator.composite'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+import io.opentracing.tag.Tag;
+import org.apiguardian.api.API;
+
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.util.Optional;
+
+import static java.util.Collections.singletonList;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Collections.singletonList'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.concurrent.CompletableFuture'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+import static org.springframework.http.HttpStatus.SERVICE_UNAVAILABLE;
+import static org.springframework.http.MediaType.TEXT_PLAIN_VALUE;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java]: Codacy Issue found: [Avoid unused imports such as 'org.springframework.web.bind.annotation.RequestMethod.GET'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
+package org.zalando.opentracing.servlet.extension;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator;
+import io.opentracing.contrib.web.servlet.filter.decorator.ServletFilterHeaderSpanDecorator.HeaderEntry;
+import io.opentracing.tag.StringTag;
+import io.opentracing.tag.Tag;
+import org.apiguardian.api.API;
+
+import javax.annotation.Nullable;
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/main/java/org/zalando/opentracing/servlet/extension/HttpRetryAfterSpanDecorator.java]: Codacy Issue found: [Avoid unused imports such as 'javax.annotation.Nullable'](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/StandardSpanDecoratorTest.java [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+        }
+
+        @Bean
+        public ServletFilterSpanDecorator staticSpanDecorator() {
+            return new StaticSpanDecorator(singletonMap("test", "true"));
+        }
+
+    }
+
+    private final RestTemplate client = new RestTemplate();
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/StandardSpanDecoratorTest.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+                return "Later";
+            } catch (final InterruptedException e) {
+                return "Interrupted";
+            }
+        };
+    }
+
+    @GetMapping("/error")
+    @ResponseStatus(SERVICE_UNAVAILABLE)
+    public void error() {
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java]: Codacy Issue found: [Document empty method body](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

File: opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (whiskeysierra) [inline opentracing-servlet-extension/opentracing-servlet-extension/src/test/java/org/zalando/opentracing/servlet/extension/ExampleController.java]: Codacy Issue found: [Document empty method body](https://app.codacy.com/app/whiskeysierra/opentracing-toolbox/pullRequest?prid=4625669)

**Comment by whiskeysierra** [comment]
I had to rebase because of the merge conflict within `<modules>`.

**Comment by whiskeysierra** [comment]
Added workaround for flaky tests

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**